File: client-kotlin/app/src/main/java/ua/meugen/android/rxjavatests_client/presenter/injections/AppComponent.java
package ua.meugen.android.rxjavatests_client.presenter.injections;

import javax.inject.Singleton;

import dagger.Component;
import ua.meugen.android.rxjavatests_client.presenter.MainPresenter;
import ua.meugen.android.rxjavatests_client.view.activities.MainActivity;

/**
 * @author meugen
 */
@Singleton
@Component(modules = { AppModule.class })
public interface AppComponent {

    MainPresenter createMainPresenter();
}

File: client-kotlin/app/src/main/java/ua/meugen/android/rxjavatests_client/presenter/injections/AppComponent.kt
package ua.meugen.android.rxjavatests_client.presenter.injections

import dagger.Component
import ua.meugen.android.rxjavatests_client.presenter.MainPresenter
import javax.inject.Singleton

/**
 * @author meugen
 */
@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {

    fun createMainPresenter(): MainPresenter
}
